### DISCUSSION TITLE
Codespaces port forwarding stopped working

### DISCUSSION THREAD

**Discussion by aaronrubinstein** [community/community#46468] [Codespaces] [answered]
### Select Topic Area

Question

### Body

As of a few hours ago port forwarding stop working for me (using the web version of VS code). When I run `npm run dev` typically I'll see a pop-up in VS code and the forwarded port will also appear under the 'ports' tab, but now neither happen. When I click the link shown in the terminal I get a 502 bad gateway message.

I tried opening a different repository in a new codespace instance and am having he same issue there as well.

![image](https://user-images.githubusercontent.com/40306940/216878052-a6e7ff39-8a73-4432-ab7c-288312c45b73.png)

![image](https://user-images.githubusercontent.com/40306940/216878105-d4534727-82f6-4ab6-b0c9-de6e548381e9.png)

**Comment by eljog** [comment]
@aaronrubinstein thank you for reporting this issue. We will take a look at this issue and get back to you.

**Comment by aaronrubinstein** [comment, marked as answer]
Per the suggestion [here](https://github.com/microsoft/vscode/issues/167655), I added `--host` to the end of my dev script in `package.json` and that solved the problem. Very strange as port forwarding had been working fine without `--host` up until yesterday (I double checked the package.json files in previous Vite projects I built in codespaces and none of them had that). Something must have changed.

![image](https://user-images.githubusercontent.com/40306940/217153866-86650a4b-87fe-4616-8c3a-c9eb3b618066.png)

**Reply by AliasHolmes** [reply]
I think Vite has some problem serving from a remote machine. I had the same issue; adding the `--host` flag did the magic.

**Reply by skifli** [reply]
Thanks, this solved the problem for me as well.

**Reply by vincentmaxwell** [reply]
If anyone else has issues with a school account where they are using Codespaces at school behind a firewall, instead of asking your IT department to open ports or whitelist the tunnels api for Visual Code Studio, try this trick instead. This made my port forwarding magically work!

**Comment by MarkBennett** [comment]
I use a Codespace for a Gatsby project and can confirm that the standard:

```
yarn gatsby develop
```

Will cause the connection to be refused and the 502 nginx bad gateway error.

Running with:

```
yarn gatsby develop -H 0.0.0.0
```

And using the port from the port list seems to solve this issue.

**Comment by benny-png** [comment]
Just use port-forwarding services like NGROK lol SUPER easy three line of shell commands https://ngrok.com/download
you can use this for reference https://youtu.be/AhiXVaWst24

**Comment by lihugang** [comment]
For me, changing to a new port for forwarding solves the problem.

**Comment by AlexYuleUSDS** [comment]
We are still having this issue -- changing the port visibility to private again seems to fix it but very annoying
